### ISSUE TITLE
Use a single named logger instance for all invalid API usages

### ISSUE THREAD

**Issue by jkwatson** [open-telemetry/opentelemetry-java#2093]
Currently, we silently fail invalid usages of the API as no-ops. It would be very useful for debugging to have a single named Logger that we can use to log all invalid API usages. It should not be enabled by default (maybe log at FINE or FINEST level), but its name should be publicly documented.

This is preferable to having individual loggers for each piece of the API, because that would entail having developers to have to manage many different logger instances, rather than having one central way to enable or disable this debugging information.

**Comment by Oberon00** [comment]
Loggers can be configured by namespace. E.g. you can set the level of everything under `io.opentelemetry.api` and `.sdk` with one line each.

**Comment by jkwatson** [comment]
> > This is preferable to having individual loggers for each piece of the API, because that would entail having developers to have to manage many different logger instances,
> 
> Loggers can be configured by package name (prefix). E.g. you can set the level of everything under `io.opentelemetry.api` and `.sdk` with one line each.

Sure, but that's likely to get a bunch of other things that are unrelated to API usage problems (for example, exporter or span processor logging in the SDK).

**Comment by gcatanese** [comment]
@jkwatson Hi John, I would like to try and help, can you please point me to an example of invalid usage you have mentioned?

**Comment by jkwatson** [comment]
> @jkwatson Hi John, I would like to try and help, can you please point me to an example of invalid usage you have mentioned?

One example would be passing `null` into any of the API methods that do not allow nulls.

**Comment by gcatanese** [comment]
@jkwatson Ok, thanks, this is what I propose (given my first contribution I would rather check I understand the problem and follow the approach before creating a PR)

Create a shared logger in `DefaultOpenTelemetry`
```     
   public static final Logger NO_OPS_LOGGER = Logger.getLogger("NoOpsLogging");
```
**Alternative**: If the public logger is not a desirable solution I would still suggest having a shared constant for the logger name ("NoOpsLogging") while creating the Logger in each class

Log as finest when needed
```
  if (context == null) {     
      NO_OPS_LOGGER.finest("Span.fromContext: context is null");    
      return Span.getInvalid();
    }
```

Logging is enabled only on-demand (`logging.level.NoOpsLogging=TRACE`) 

Let me know what you think (and if you agree with any useful info/guidelines would be appreciated - I imagine I would have to 'touch' several classes)

**Comment by jkwatson** [comment]
I don't think that name is correct. We want one logger for all API mis-uses, not just for "no ops". This will include baggage and context mis-use as well, neither of which are no-op in the API.

**Comment by gcatanese** [comment]
This can now be closed I believe (see PR #4073)
